### PR TITLE
fix(svelte): cannot parse action at /session

### PR DESCRIPTION
**Pull request by bwallberg**
## ☕️ Reasoning

All frameworks should rely on the core util `createActionsURL`

## 🧢 Checklist

- [ ] Documentation
- [x] Tests
- [x] Ready to be merged

## 🎫 Affected issues

Depends on #10094 to solve the relevant issue.

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Review by ndom91 — APPROVED** [review]

File: packages/frameworks-sveltekit/src/lib/actions.ts [hunk 1]
```diff
@@ -78,11 +82,11 @@ export async function signOut(
   config: SvelteKitAuthConfig,
   event: RequestEvent
 ) {
-  const { request } = event
+  const { request, url: { protocol } } = event
   const headers = new Headers(request.headers)
   headers.set("Content-Type", "application/x-www-form-urlencoded")
 
-  const url = createActionURL("signout", headers, config.basePath)
+  const url = createActionURL("signout", protocol, headers, process.env, config.basePath)
```
Comment (ThangHuuVu) [inline packages/frameworks-sveltekit/src/lib/actions.ts:89]: for SvelteKit we should use the env object from [`"$env/dynamic/private"`](https://kit.svelte.dev/docs/modules#$env-dynamic-private) - that is SvelteKit's abstraction to access env vars from different adapters
```suggestion
  const url = createActionURL("signout", protocol, headers,env, config.basePath)
```

File: packages/frameworks-sveltekit/src/lib/actions.ts [hunk 2]
```diff
@@ -105,9 +109,9 @@ export async function auth(
   setEnvDefaults(env, config)
   config.trustHost ??= true
 
-  const { request: req } = event
+  const { request: req, url: { protocol } } = event
 
-  const sessionUrl = createActionURL("session", req.headers, config.basePath)
+  const sessionUrl = createActionURL("session", protocol, req.headers, process.env, config.basePath)
```
Comment (ThangHuuVu) [inline packages/frameworks-sveltekit/src/lib/actions.ts:114]:
```suggestion
  const sessionUrl = createActionURL("session", protocol, req.headers,env, config.basePath)
```

File: packages/frameworks-sveltekit/src/lib/actions.ts [hunk 3]
```diff
@@ -26,7 +24,13 @@ export async function signIn(
   } = options instanceof FormData ? Object.fromEntries(options) : options
 
   const callbackUrl = redirectTo?.toString() ?? headers.get("Referer") ?? "/"
-  const base = createActionURL("signin", headers, config.basePath)
+  const base = createActionURL(
+    "signin",
+    protocol,
+    headers,
+    process.env,
```
Comment (ThangHuuVu) [inline packages/frameworks-sveltekit/src/lib/actions.ts:31]:
```suggestion
   env,
```

**Review by ThangHuuVu — CHANGES_REQUESTED** [review]

File: packages/frameworks-sveltekit/src/lib/actions.ts [hunk 4]
```diff
@@ -26,7 +24,13 @@ export async function signIn(
   } = options instanceof FormData ? Object.fromEntries(options) : options
 
   const callbackUrl = redirectTo?.toString() ?? headers.get("Referer") ?? "/"
-  const base = createActionURL("signin", headers, config.basePath)
+  const base = createActionURL(
+    "signin",
+    protocol,
+    headers,
+   env,
```
Comment (ThangHuuVu) [inline packages/frameworks-sveltekit/src/lib/actions.ts:31]:
```suggestion
    env,
```

File: packages/frameworks-sveltekit/src/lib/actions.ts [hunk 5]
```diff
@@ -78,11 +82,11 @@ export async function signOut(
   config: SvelteKitAuthConfig,
   event: RequestEvent
 ) {
-  const { request } = event
+  const { request, url: { protocol } } = event
   const headers = new Headers(request.headers)
   headers.set("Content-Type", "application/x-www-form-urlencoded")
 
-  const url = createActionURL("signout", headers, config.basePath)
+  const url = createActionURL("signout", protocol, headers,env, config.basePath)
```
Comment (ThangHuuVu) [inline packages/frameworks-sveltekit/src/lib/actions.ts:89]:
```suggestion
  const url = createActionURL("signout", protocol, headers, env, config.basePath)
```

**Review by ThangHuuVu — APPROVED** [review]

**Comment by aakash14goplani** [comment]
@ThangHuuVu and @ndom91 
The recent upgrade i.e. 0.14.1 broke my application

Here is my configuration that was working fine till 0.14.0
- NO Authjs specific environment variables set in .env file
- High level config object:
   ```ts
   export const { handle: getAuthConfig } = SvelteKitAuth(async (event) => {
     const useSecureCookies = !dev;
     const config: SvelteKitAuthConfig = {
       providers: [
        { ... }
       ],
       basePath: '/my-app/auth',
       pages: {
         error: '/server-auth-error'
       }
     }
   };
   ```
- In svelte.config, it is set as
   ```ts
   kit: {
     adapter: adapter(),
     paths: {
       base: '/my-app',
       relative: false
     }
   }
   ```

After upgrade, I am getting error:
> Error caught in [server-hooks]: Invalid URL TypeError: Invalid URL
    at new URL (node:internal/url:775:36)
    at Module.createActionURL (file:/node_modules/.pnpm/@auth+core@0.28.1/node_modules/@auth/core/lib/utils/env.js:66:15)
    at Module.auth (\node_modules\.pnpm\@auth+sveltekit@0.14.1_@sveltejs+kit@2.5.5_svelte@4.2.12\node_modules\@auth\sveltekit\dist\actions.js:69:24)
    at event.locals.auth (\node_modules\.pnpm\@auth+sveltekit@0.14.1_@sveltejs+kit@2.5.5_svelte@4.2.12\node_modules\@auth\sveltekit\dist\index.js:323:41)
    at userSessionInterceptor (src\hooks.server.ts:24:37)
    at apply_handle (node_modules\.pnpm\@sveltejs+kit@2.5.5_@sveltejs+vite-plugin-svelte@3.0.2_svelte@4.2.12_vite@5.2.7\node_modules\@sveltejs\kit\src\exports\hooks\sequence.js:85:11)
    at resolve (node_modules\.pnpm\@sveltejs+kit@2.5.5_@sveltejs+vite-plugin-svelte@3.0.2_svelte@4.2.12_vite@5.2.7\node_modules\@sveltejs\kit\src\exports\hooks\sequence.js:110:9)
    at handle (node_modules\.pnpm\@auth+sveltekit@0.14.1_@sveltejs+kit@2.5.5_svelte@4.2.12\node_modules\@auth\sveltekit\dist\index.js:334:20)
    at async Module.respond (\node_modules\.pnpm\@sveltejs+kit@2.5.5_@sveltejs+vite-plugin-svelte@3.0.2_svelte@4.2.12_vite@5.2.7\node_modules\@sveltejs\kit\src\runtime\server\respond.js:319:20)
    at async file:/node_modules/.pnpm/@sveltejs+kit@2.5.5_@sveltejs+vite-plugin-svelte@3.0.2_svelte@4.2.12_vite@5.2.7/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:524:22 {
  code: 'ERR_INVALID_URL',
  input: 'http:://localhost:4200'

What am I missing here?

**Comment by ndom91** [comment]
@aakash14goplani this is unrelated, but you should have an `AUTH_SECRET` env var set or a `secret` configuration option.

Regarding your error, looks like its an actual issue. Someone submitted a PR for it here: https://github.com/nextauthjs/next-auth/pull/10421/files

We'll merge it soon :pray:

**Pull request merged**